src/ConfigEditor/ConfigEditor.test.tsx

```diff
@@ -51,287 +52,309 @@ jest.mock('@grafana/runtime', () => {
+        await select(selectEl, workgroup.workgroupName, { container: document.body });
+
+        await waitFor(() => expect(onChange).toHaveBeenCalledTimes(1));
+        expect(onChange).toHaveBeenCalledWith({
           ...props.options,
-          jsonData: { ...props.options.jsonData, database: 'test-db' },
-        }}
-        onOptionsChange={onChange}
-      />
-    );
-
-    const selectEl = screen.getByLabelText(selectors.components.ConfigEditor.ClusterID.input);
-    expect(selectEl).toBeInTheDocument();
-    await select(selectEl, cluster.clusterIdentifier, { container: document.body });
-
-    await waitFor(() => expect(onChange).toHaveBeenCalledTimes(1));
-    expect(onChange).toHaveBeenCalledWith({
-      ...props.options,
-      url: 'bar.d.e.f:456/test-db',
-      jsonData: { ...props.options.jsonData, database: 'test-db', clusterIdentifier: clusterIdentifier },
-    });
-  });
-
-  it('should update an existing url when specifying a database', async () => {
-    const onChange = jest.fn();
-    await act(async () => {
-      render(
-        <ConfigEditor
-          {...props}
-          onOptionsChange={onChange}
-          options={{
-            ...props.options,
-            url: 'my.cluster.address:123/my-old-db',
-            jsonData: { ...props.options.jsonData, clusterIdentifier, useServerless: false, useManagedSecret: false },
-          }}
-        />
-      );
-    });
-
-    const dbField = screen.getByTestId('data-testid database');
-    expect(dbField).toBeInTheDocument();
-    fireEvent.change(dbField, { target: { value: 'abcd' } });
-
-    expect(onChange).toHaveBeenCalledTimes(1);
-    expect(onChange).toHaveBeenCalledWith({
-      ...props.options,
-      // the endpoint is updated as re-fetched
-      url: 'bar.d.e.f:456/abcd',
-      jsonData: {
-        ...props.options.jsonData,
-        clusterIdentifier,
-        useServerless: false,
-        useManagedSecret: false,
-        database: 'abcd',
-      },
-    });
-  });
-
-  it('should show the cluster identifier and the db user', async () => {
-    const onChange = jest.fn();
-    render(
-      <ConfigEditor
-        {...props}
-        onOptionsChange={onChange}
-        // setting the managedSecret will trigger the secret retrieval
-        options={{
+          url: 'foo.a.b.c:123/test-db',
+          jsonData: { ...props.options.jsonData, database: 'test-db', workgroupName },
+        });
+      });
+
+      it('should populate the `url` prop when clusterIdentifier is selected', async () => {
+        const onChange = jest.fn();
+        render(
+          <ConfigEditor
+            {...props}
+            options={{
+              ...props.options,
+              jsonData: { ...props.options.jsonData, database: 'test-db', useManagedSecret: false },
+            }}
+            onOptionsChange={onChange}
+          />
+        );
+
+        const selectEl = screen.getByRole('combobox', { name: selectors.components.ConfigEditor.ClusterID.input });
```

Comment:
I think the old test was selecting the wrong Cluster identifier input. As it was set up, cluser Id Select shouldn't have been shown, instead the cluster disabled autofilled field should have.
I added a prop to get it to show, and a selector for the combobox in particular (not the input).
I could be getting this wrong, but I think it should be ok now. I double checked https://github.com/grafana/redshift-datasource/pull/228 to verify the logic.